src/cmr/sizing/formats.clj

```diff
@@ -63,16 +80,42 @@
     (+ (* granule-count compression measurements)
        (* granule-count metadata))))
 
-(defn- estimate-netcdf4-or-ascii-size
-  "Calculates the estimated size for ASCII or NETCDF4 format.
+(defn- estimate-ascii-size
+  "Calculates the estimated size for ASCII.
+  In ASCII output, the actual size for a granule in bytes is the sum of digit number
+  of all the dimension data + 2 bytes(comma and space) for each data except for the last one.
+  To estimate, we assume 50% dimension data has the average FillValues digit number.
+  the other 50% dimention data has the average ValidRanges digit number.
+  The estimated size = 0.5 * avg-fill-value-digit-number * dimensionality +
+                       0.5 * avg-valid-range-digit-number * dimensionality +
+                       2 * (dimensionality -1)"
+  [granule-count variables params]
+  (* granule-count  
+     (reduce (fn [total-estimate variable]
+               (let [dimensionality (get-dimensionality variable) 
+                     avg-fill-value-digit-number (get-avg-fill-value-digit-number variable)
+                     avg-valid-range-digit-number (get-avg-valid-range-digit-number variable)]
+                 (log/info (format (str "request-id: %s variable-id: %s total-estimate: %s "
+                                        "dimensionality: %s avg-fill-value-digit-number: %s "
+                                        "avg-valid-range-digit-number: %s")
+                                   (:request-id params) (get-in variable [:meta :concept-id])
+                                   total-estimate dimensionality avg-fill-value-digit-number 
+                                   avg-valid-range-digit-number))
+                 (+ total-estimate
+                    (* 0.5 avg-fill-value-digit-number dimensionality)
+                    (* 0.5 avg-valid-range-digit-number dimensionality)
+                    (* 2 (- dimensionality 1))))) 
```

Comment:
Oh because there's one less comma and space. I wouldn't have thought of that. 👍 

Nit: I'd probably use `dec` rather than subtract 1 for readability.

Reply:
Yeah, in fact, the linter should have caught that ... it's one of the idioms recommended by the Clojure style guide.